### ISSUE TITLE
meshlabserver crashes if Screened Poisson Surface Reconstruction is used

### ISSUE THREAD

**Issue by obbenb** [cnr-isti-vclab/meshlab#219]
I have a very simple filter file (included at the end) that creates normals for a point cloud and then creates a surface using Screened Poisson Surface Reconstruction.

I have compiled the meshlab from github master repo  myself on Kubuntu 16.10. I have also tried using zarquon42's Ubuntu 16.04 package at https://launchpad.net/~zarquon42/+archive/ubuntu/meshlab/+packages (albeit installed to Kubuntu 16.10). Both of them have  the same error when I use meshlabserver. The filter script runs in Meshlab GUI without any issue.

At the beginning, I see the error: 

_Error While parsing the XML filter plugin descriptors: We are trying to load a xml file that does not correspond to any dll or javascript code; please delete all the spurious xml files_
then later
_LOG: 1 WARNING! The MeshLab Script System is able to manage just the C++ XML filters._

GDB shows below error:

_Reading filter with name Screened Poisson Surface Reconstruction
Starting Script of 2 actionsfilter: Compute normals for point sets
Building KdTree...filter: Screened Poisson Surface Reconstruction
LOG: 1 WARNING! The MeshLab Script System is able to manage just the C++ XML filters.
Problem with filter: Screened Poisson Surface Reconstruction
Failed to apply script file /..../convert.mlx
[Thread 0x7fffdb1f6700 (LWP 5552) exited]

The 
Thread 2 "QXcbEventReader" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffec068700 (LWP 5550)]
0x00007fffefa34299 in ?? ()_

stack trace is not helpful, only showing above thread addresses. I do not think QXcbEventReader is the issue, but caused by the crash on thread 0 not  letting  QT to close application cleanly.  I have tried without surface reconstruction plugin, and it works fine without crashing. So issue seems to me in C++ plugin. 

command line parameters are below (I have also tried adding -m vc vf vq vn vt fc ff fq fn wc wn wt)
 _-i "1.obj" -o "test.ply" -s convert.mlx_

There are related  xml files in plugins directory, and strace shows the shared library src/distrib/plugins/libfilter_screened_poisson.so is loaded.

I can not us snap package since meshlabserver there crashes  too with _"no additional memory available!!! memory required: 4048704"_ message.

I appreciate any help with directions to investigate.

Thx

convert.mlx 

_<!DOCTYPE FilterScript>
<FilterScript>
 <filter name="Compute normals for point sets">
  <Param type="RichInt" value="10" description="Neighbour num" tooltip="The number of neighbors used to estimate normals." name="K"/>
  <Param type="RichInt" value="0" description="Smooth Iteration" tooltip="The number of smoothing iteration done on the p used to estimate and propagate normals." name="smoothIter"/>
  <Param type="RichBool" value="false" description="Flip normals w.r.t. viewpoint" tooltip="If the 'viewpoint' (i.e. scanner position) is known, it can be used to disambiguate normals orientation, so that all the normals will be oriented in the same direction." name="flipFlag"/>
  <Param type="RichPoint3f" y="0" z="0" x="0" description="Viewpoint Pos." tooltip="The viewpoint position can be set by hand (i.e. getting the current viewpoint) or it can be retrieved from mesh camera, if the viewpoint position is stored there." name="viewPos"/>
 </filter>
 <xmlfilter name="Screened Poisson Surface Reconstruction">
  <xmlparam value="0" name="cgDepth"/>
  <xmlparam value="false" name="confidence"/>
  <xmlparam value="4" name="depth"/>
  <xmlparam value="5" name="fullDepth"/>
  <xmlparam value="8" name="iters"/>
  <xmlparam value="4" name="pointWeight"/>
  <xmlparam value="false" name="preClean"/>
  <xmlparam value="1.5" name="samplesPerNode"/>
  <xmlparam value="1.1" name="scale"/>
  <xmlparam value="false" name="visibleLayer"/>
 </xmlfilter>
</FilterScript>_

**Comment by parisana** [comment]
I got the same issue,
```
Error While parsing the XML filter plugin descriptors: We are trying to load a xml file that does not correspond to any dll or javascript code; please delete all the spurious xml files
Error While parsing the XML filter plugin descriptors: We are trying to load a xml file that does not correspond to any dll or javascript code; please delete all the spurious xml files
Error While parsing the XML filter plugin descriptors: We are trying to load a xml file that does not correspond to any dll or javascript code; please delete all the spurious xml files
Total 244 filtering actions
Total 12 io plugins
Opening a file with extention obj
```

**Comment by WenbinFei** [comment]
Hi Obbenb,

Have you find any solution, I got the same issue.

Best,

Wenbin

**Comment by themancalledjakob** [comment]
This is due to xml filters without dylib.
check here:
https://simoncblyth.bitbucket.io/env/notes/graphics/meshlab/meshlab_build/
or here (line 377):
https://github.com/cnr-isti-vclab/meshlab/blob/master/src/common/pluginmanager.cpp

For me it is a non-fatal error and meshlab/meshlabserver just runs as usual, also if I use Screen Poisson Surface Reconstruction.
So I guess it is not necessarily connected to your issue.

**Comment by alemuntoni** [comment]
This issue should be solved in the latest version (2020.06). Feel free to reopen if the problem persists.
